docs/deprecated/clusters/public-cloud/azure/required-permissions.md

```diff
@@ -0,0 +1,967 @@
+             "Microsoft.Network/privateDnsZones/read",
+             "Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete",
+             "Microsoft.Network/privateDnsZones/virtualNetworkLinks/read",
+             "Microsoft.Network/privateDnsZones/virtualNetworkLinks/write",
+             "Microsoft.Network/privateDnsZones/write",
+             "Microsoft.Network/publicIPAddresses/delete",
+             "Microsoft.Network/publicIPAddresses/join/action",
+             "Microsoft.Network/publicIPAddresses/read",
+             "Microsoft.Network/publicIPAddresses/write",
+             "Microsoft.Network/routeTables/delete",
+             "Microsoft.Network/routeTables/read",
+             "Microsoft.Network/routeTables/write",
+             "Microsoft.Network/virtualNetworks/delete",
+             "Microsoft.Network/virtualNetworks/join/action",
+             "Microsoft.Network/virtualNetworks/join/action",
+             "Microsoft.Network/virtualNetworks/joinLoadBalancer/action",
+             "Microsoft.Network/virtualNetworks/peer/action",
+             "Microsoft.Network/virtualNetworks/read",
+             "Microsoft.Network/virtualNetworks/subnets/delete",
+             "Microsoft.Network/virtualNetworks/subnets/join/action",
+             "Microsoft.Network/virtualNetworks/subnets/joinLoadBalancer/action",
+             "Microsoft.Network/virtualNetworks/subnets/read",
+             "Microsoft.Network/virtualNetworks/subnets/virtualMachines/read",
+             "Microsoft.Network/virtualNetworks/subnets/write",
+             "Microsoft.Network/virtualNetworks/virtualMachines/read",
+             "Microsoft.Network/virtualNetworks/virtualNetworkPeerings/delete",
+             "Microsoft.Network/virtualNetworks/virtualNetworkPeerings/read",
+             "Microsoft.Network/virtualNetworks/virtualNetworkPeerings/write",
+             "Microsoft.Network/virtualNetworks/write",
+             "Microsoft.Resources/subscriptions/resourceGroups/read",
+             "Microsoft.Storage/storageAccounts/blobServices/containers/read",
+             "Microsoft.Storage/storageAccounts/blobServices/containers/write",
+             "Microsoft.Storage/storageAccounts/read",
+             "Microsoft.Storage/storageAccounts/write"
+           ],
+           "NotActions": [],
+           "AssignableScopes": ["/subscriptions/$SUBSCRIPTION_ID"]
+         }
+         ```
+
+       </details>
+
+3.  Create the custom role, _Dynamic Placement Palette Deployer_, in Azure using the JSON file you created in the
+    previous step. Issue the following command to create the role.
+
+    ```shell
+    az role definition create --role-definition @dynamic-permissions.json --output table
+    ```
+
+4.  Export the
+    [security principal](https://learn.microsoft.com/en-us/azure/role-based-access-control/overview#security-principal)
+    object ID you want to use with Palette to a variable.
+
+    ```shell
+    export ASSIGNEE="<security_principal_object_id>"
```

Comment:
this needs to say `<security_principal_client_id>` instead of `<security_principal_object_id>`

Reply:
Thanks @kreeuwijk. @mattwelke and I tested the flow using an application's client ID and it worked. So we'll go with your recommendation 👍🏻